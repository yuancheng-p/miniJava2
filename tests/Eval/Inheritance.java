public class Inheritance {

  public static void main (String [] args) {

    B b = new B();
    int i = b.getNumberA();
    int j = b.getNumberB();
  }
}

class A {
  int a = 10;
  int getNumberA () {
    return a;
  }
}


class B extends A {
  int k = 20;
  int getNumberB () {
    return k;
  }
}

