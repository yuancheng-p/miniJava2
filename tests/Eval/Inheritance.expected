successfull parsing
start typing
check add and methods:Inheritance:main
check add and methods:A:getNumberA
check add and methods:B:getNumberB
--------- env --------
A : class_env(parent=: Object)
method : getNumberA
 attr:  int a = 10;
B : class_env(parent=: A)
method : getNumberB
 attr:  int k = 20;
Boolean : class_env(parent=: Object)
Inheritance : class_env(parent=: Object)
method : main paramlist: String[] 
Integer : class_env(parent=: Object)
Object : class_env(parent=: )
String : class_env(parent=: Object)
end typing
_______________main___________________
B b = new B();
    /*
    ------ b in frame ------
    b : 0 (ref)
    ------ object in heap ------
    [B] (0) :
      a : 10
      k : 20
    */

int i = b.getNumberA();
    /*
    ------ Call method: getNumberA() ---------
    ------ Return: 10 ------------
    */

    /*
    ------ i in frame ------
    i : 10
    */

int j = b.getNumberB();
    /*
    ------ Call method: getNumberB() ---------
    ------ Return: 20 ------------
    */

    /*
    ------ j in frame ------
    j : 20
    */

